### PR TITLE
Issue 341: Demonstrate Pravega Stream and Flink auto-scaling

### LINKED ISSUE #341: Demonstrate Pravega Stream and Flink auto-scaling
**Problem description**
Create a demonstration that shows how Pravega stream can auto-scale and Flink task managers can dynamically change in parallelism according to the number of Pravega Stream segments.

### PR DESCRIPTION
**Pull request by RaulGracia**
**Change log description**
Adds a new sample that demonstrates coordinated auto-scaling of Pravega streams and Flink task managers.

**Purpose of this change**
Fixes #341.

**What the code does**
Adds a new scenario wit the applications and instructions to demonstrate coordinated auto-scaling of Pravega streams and Flink task managers. Note as it has 2 sub-projects, this sample is built independently in its own folder.

**How to verify it**
TBD.

### CODE REVIEW

File: scenarios/pravega-flink-autoscaling/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+you should be able to see a sinusoidal workload being generated by
+the `Data Generator` and the number of stream segments changing 
+as a result of the change in load and the stream scaling
+policy configured. Finally, this should lead to the Kubernetes
+`Horizontal Pod Auto-scaler` changing the number of `Flink Task Managers`
+according to the number of parallel segments in the stream.
+In the following, we provide some `Grafana` dashboard screenshots
+demonstrating this:
+
+![image](img/sinusoidal-workload.png)
```
Comment (crazyzhou) [inline scenarios/pravega-flink-autoscaling/README.md:181]: Is it good to show this image earlier in the `Data Generator` section to show the pattern of the load?

File: scenarios/pravega-flink-autoscaling/data-generator/build.gradle [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (crazyzhou) [inline scenarios/pravega-flink-autoscaling/data-generator/build.gradle:2]: Copyright should be Pravega authors

File: scenarios/pravega-flink-autoscaling/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
+# Scaling Flink automatically on K8S by monitoring Pravega metrics
+
+## Components
```
[the remark this answers is not in the document]
Comment (RaulGracia) [reply scenarios/pravega-flink-autoscaling/README.md:14]: Added links in a new README paragraph to introduce the demo.

File: scenarios/pravega-flink-autoscaling/README.md (hunk 1 above)
Comment (RaulGracia) [reply scenarios/pravega-flink-autoscaling/README.md:193]: I think that is better to keep the results for the last part, as we focus first on having the demo up and runnig.

File: scenarios/pravega-flink-autoscaling/data-generator/build.gradle (hunk 2 above)
Comment (RaulGracia) [reply scenarios/pravega-flink-autoscaling/data-generator/build.gradle]: I have changed this in this sample, but in the future we will need to change it in the rest of the repo.

**Comment by RaulGracia** [comment]
I have tested the current code in a k8s (see versions below) cluster and it works:
```
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.4"..."}
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.9+vmware.1"..."}
```

**Review by crazyzhou — APPROVED** [review]
LGTM

**Pull request merged**